### PR TITLE
[Core] Missing template on math_utils

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

**Comment by loumalouomega** [comment]
Ping @KratosMultiphysics/technical-committee

File: kratos/utilities/math_utils.h [hunk 1]
```diff
@@ -1360,44 +1392,79 @@ class MathUtils
...
+        TMatrixType strain_tensor;
+
+#ifdef KRATOS_USE_AMATRIX   // This macro definition is for the migration period and to be removed afterward please do not use it
         if (rStrainVector.size()==3) {
-            StrainTensor.resize(2,2, false);
-            StrainTensor(0,0) = rStrainVector[0];
-            StrainTensor(0,1) = 0.5*rStrainVector[2];
-            StrainTensor(1,0) = 0.5*rStrainVector[2];
-            StrainTensor(1,1) = rStrainVector[1];
+            strain_tensor.resize(2,2, false);
```
Comment (RiccardoRossi) [inline kratos/utilities/math_utils.h:1374]: what is the difference with and without amatrix?

**Comment by loumalouomega** [comment]
The []/()

**Comment by RiccardoRossi** [comment]
in principle ot is what you corrrcted right?

**Comment by RiccardoRossi** [comment]
the version with [] works both with AMATRIX and ublas. The version with () only works with ublas.

hence after your latest change, the AMATRIX version should be the only one (without distinction)

**Comment by loumalouomega** [comment]
With this change both version work. There are some bounded arrays that justwork with () in ublas. Trust me you can compile the StructuralMechanics without it

**Comment by loumalouomega** [comment]
I corrected

**Comment by RiccardoRossi** [comment]
there should be a single version valid both for AMatrix and ublas.

can u point out exactly what the problem is? it should be definitely possible to do without the macro

File: kratos/utilities/math_utils.h [hunk 2]
```diff
@@ -1360,44 +1392,79 @@ class MathUtils
...
+            strain_tensor(1,2) = 0.5*rStrainVector[4];
+            strain_tensor(2,0) = 0.5*rStrainVector[5];
+            strain_tensor(2,1) = 0.5*rStrainVector[4];
+            strain_tensor(2,2) = rStrainVector[2];
         }
+#else
+        if (rStrainVector.size()==3) {
+            strain_tensor.resize(2,2, false);
+            strain_tensor(0,0) = rStrainVector(0);
+            strain_tensor(0,1) = 0.5*rStrainVector(2);
```
Comment (RiccardoRossi) [inline kratos/utilities/math_utils.h]: I am sure that this works with [] in ublas.
I mean

          strain_tensor(0,0) = rStrainVector[0];

**Comment by loumalouomega** [comment]
There is a bounded array that can be accessed only via () in ublas, try to
compile the StructuralMechanicsApplication without it and you won't be
able.

**Comment by RiccardoRossi** [comment]
both array_1d and Vector provide [] access. You should not be using anyhting more than that...

**Comment by RiccardoRossi** [comment]
i just pushed a solution for your problem.

the thing is that you guys are in my opinion abusing expressions (and you definitely do not want to pass expressions into the function)

**Comment by loumalouomega** [comment]
Thanks, need approval BTW

**Review by RiccardoRossi — DISMISSED** [review]

**Comment by loumalouomega** [comment]
Solving conflict

**Comment by loumalouomega** [comment]
@RiccardoRossi this is ready

**Comment by loumalouomega** [comment]
Anyone?

**Comment by loumalouomega** [comment]
@roigcarlo can you approve please?

**Review by roigcarlo — APPROVED** [review]
Reaproving. 

Also, notice that some "merge with master" no longer require re approval, its done automatically

**Comment by loumalouomega** [comment]
@RiccardoRossi We finally solved the problem without using the macro

**Pull request merged**